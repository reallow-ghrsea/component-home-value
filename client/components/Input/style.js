export const InputContainer = styled.div`
    display: flex;
    width: 100%;
    justify-content: center;
    padding: 0;
    border: 1px solid #ccc;
    background-color: #fff;

`;

export const InputBorder = styled.div`
    width: 92%;
    display: flex;
    justify-content: center;
    margin: 0 0 0 60px;
    padding: 5px;
    background-color: #f2f2f2;
`;

export const ReplyInput = styled.input`
    width: 100%;
    height: 20px;
    border-style: none;
    margin-left: 5px;
    &:focus {
    outline: none;
    }
    &:placeholder: {
        color: #ccc;
    }
`;

export const User = styled.a`
    font-size: 12px;
    display: flex;
    align-self: center;
    font: 12px/1.4 "Lucida Grande","Lucida Sans Unicode","Lucida Sans",Garuda,Verdana,Tahoma,sans-serif;
    color: #38d;
    padding: 3px;
    
    &: hover {
        color: #333;
    }
    ${props => props.deleting === false && `
        color: #fff;
        background: #06c;
        letter-spacing: 0.1px;
    `}
`;


export const ProfilePicture = styled.span`
    background-position: center;
    width: 20px;
    height: 20px;
    border: 1px solid #ccc;
    background-image: url(${props => (props.image ? props.image : 'linear-gradient(135deg,#846170,#70929c)')});
`;
