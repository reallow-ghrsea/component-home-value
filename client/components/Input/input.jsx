import React, { useState, useEffect } from 'react';
import * as Styles from './style';


const Input = (props) => {
  const [value, valueChange] = useState('');
  const [username, removeUsername] = useState(null);
  let replyInput;

  useEffect(() => {
    replyInput.focus();
  });

  const removeUser = (e) => {
    if (!username) {
      if (value === '' && e.key === 'Backspace') {
        if (username === null) {
          removeUsername(false);
        } else if (username === false) {
          removeUsername(true);
        }
      } else if (value !== '') {
        removeUsername(null);
      }
    }
  };

  return (
    <Styles.InputBorder>
      <Styles.InputContainer>
        <Styles.ProfilePicture image={props.image} />
        {!username
        && <Styles.User deleting={username}>{`@${props.username}`}</Styles.User>
      }
        <Styles.ReplyInput className="replyInputKevin" ref={(input) => { replyInput = input; }} value={value} onKeyUp={e => removeUser(e)} onChange={e => valueChange(e.target.value)} placeholder="Write a Reply" />
      </Styles.InputContainer>
    </Styles.InputBorder>
  );
};

export default Input;
