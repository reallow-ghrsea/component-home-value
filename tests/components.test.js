import React from 'react';
import { configure, mount, shallow } from 'enzyme';
import { cleanup } from 'react-testing-library';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import data from './testData';
import Count from '../client/components/CommentCount/commentCount';
import Input from '../client/components/Input/input';
import Profile from '../client/components/ProfilePicture/profilePicture';

configure({ adapter: new Adapter() });
beforeEach(cleanup);


describe('Count Component tests', () => {
  it('01- Matches Snapshot', () => {
    const props = { count: 100 };
    const wrapper = shallow(<Count {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('02- Text Matches Props', () => {
    const props = { count: 100 };
    const wrapper = shallow(<Count count={100} />);
    expect(wrapper.find('.commentCount').text()).toEqual('100 comments');
  });
});


describe('Input Component tests', () => {
  it('01- Matches Snapshot', () => {
    const props = { username: 'Test', image: data[0].profilePic };
    const wrapper = shallow(<Input {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('02- input value changes', () => {
    const props = { username: 'Test', image: data[0].profilePic };
    const wrapper = mount(<Input {...props} />);
    expect(toJson(wrapper.find('input')).props.value).toEqual('');
    wrapper.find('input').simulate('change', { target: { value: 'test' } });
    expect(toJson(wrapper.find('input')).props.value).toEqual('test');
  });


  describe('Profile Component tests', () => {
    it('01- Matches Snapshot', () => {
      const props = { image: data[0].image, followers: 100, username: 'Test' };
      const wrapper = mount(<Profile {...props} />);
    });

    it('02- Profile renders on hover', () => {
      const props = { image: data[0].image, followers: 100, username: 'Test' };
      const wrapper = mount(<Profile {...props} />);
      wrapper.simulate('mouseEnter');
      expect(toJson(wrapper.find('.profileContainer'))).not.toBe(null);
      wrapper.simulate('mouseleave');
      expect(toJson(wrapper.find('.profileContainer'))).toBe(null);
    });
  });
});
