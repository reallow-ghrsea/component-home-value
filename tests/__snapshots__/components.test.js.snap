// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Count Component tests 01- Matches Snapshot 1`] = `
<styled.div>
  <FontAwesomeIcon
    border={false}
    className=""
    fixedWidth={false}
    flip={null}
    icon="comment-alt"
    inverse={false}
    listItem={false}
    mask={null}
    pull={null}
    pulse={false}
    rotation={null}
    size={null}
    spin={false}
    style={
      Object {
        "color": "#999",
      }
    }
    symbol={false}
    title=""
    transform={null}
  />
  <styled.span
    className="commentCount"
  >
    100 comments
  </styled.span>
</styled.div>
`;

exports[`Input Component tests 01- Matches Snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.span
      image="https://s3.us-east-2.amazonaws.com/kevinbece5/userpics/user2.jpg"
    />
    <styled.a
      deleting={null}
    >
      @Test
    </styled.a>
    <styled.input
      className="replyInput"
      onChange={[Function]}
      onKeyUp={[Function]}
      placeholder="Write a Reply"
      value=""
    />
  </styled.div>
</styled.div>
`;
